pkg/student/student.go

```diff
@@ -26,7 +32,34 @@ func NewRouter(s rest.Server, store postgres.StudentStore) *chi.Mux {
 	})
 	return r
 }
+func  authorizationMiddleware(s rest.Server,store postgres.StudentStore) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return s.NewHandler(func(w http.ResponseWriter, r *http.Request) *rest.Error {
+			studentId := chi.URLParam(r, "studentId")
+
+			// Verify user access to the school
+			session, ok := auth.GetSessionFromCtx(r.Context())
+			if !ok {
+				return auth.NewGetSessionError()
+			}
+			student, err := store.Get(studentId)
```

Comment:
I think this could be simplified to a single call to CheckPermission. CheckPermission can be modified to receive studentId and userId as parameter. Then it will query everything in one go, something like this?
```go
func (s StudentStore) CheckPermissions(studentId string ,userId string) (bool,error){
    var student Student
    if err:=s.Model(&student).
        Relation("School").
        Relation("Users").
        Where("student.id=?",studentId).
        Where("user.id=?",userId).
        Select();err == pg.ErrNoRows {
        return false, nil
    } else if err!=nil{
        return false, err
    }
    return false,err
}
```
See [go-pg relation docs](https://github.com/go-pg/pg/wiki/Writing-Queries#has-one).

Reply:
Can't do that. SInce student doesn't have any relation to users. Might need to check on other solutions

Reply:
Oh right, try relation(“School.Users”) instead

Reply:
That will tell go-pg to automatically resolve the relationship between school with user, using the userToSchool table.

Reply:
Cannot. Is this possible to be done using one query?

Reply:
> `SELECT * FROM public.students as s
JOIN public.user_to_schools as p
ON s.school_id=p.school_id
WHERE s.id='fab46b76-0d55-487c-9625-de21c947f757' AND p.user_id='48180cd0-6203-4048-9501-3c3bbbe79a74'

`